pkg/node/fsm/wait_micro_snapshot.go

```diff
@@ -162,8 +162,21 @@ func (a *WaitMicroSnapshotState) checkAndAppendMicroBlock(
 	if err != nil {
 		return nil, err
 	}
-	// TODO: check if light node feature activated + 1000 blocks
-	newBlock.StateHash = micro.StateHash
+
+	blockchainHeight, err := a.baseInfo.storage.Height()
+	if err != nil {
+		return nil, a.Errorf(errors.Wrap(err, "failed to get blockchain height"))
+	}
+	// here the blockchainHeight is equal to lastBlockHeight because we are appending a microblock to the last block
+	lastBlockHeight := blockchainHeight
+	ok, err = a.baseInfo.storage.IsActiveLightNodeNewBlocksFields(lastBlockHeight)
+	if err != nil {
+		return nil, a.Errorf(err)
+	}
+	if ok {
+		newBlock.StateHash = micro.StateHash
+	}
+
```

Comment:
Now I think that all these logic is not necessary because the `StateHash` field existence should be checked in `ApplyMicroWithSnapshots`